### PR TITLE
Issue #487: Support more data in .info files

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop-ops/backdropcms.org/issues/487

Includes fix for API example.

### CODE REVIEW

File: www/modules/custom/borg_github/borg_github.module [hunk 1]
```diff
@@ -8,15 +8,189 @@
...
+          'Authorization' => 'token ' . $access_token,
+          'Accept' => 'application/vnd.github.v3.raw',
+        ),
+      ));
+      if ($contents_result->code == 200) {
+        $parts = pathinfo($screenshot_url);
+        $filename = file_munge_filename($parts['basename'], 'png jpg');
+        $uri = 'public://theme-screenshots/' . $payload['repository']['name'] . '/' . $parts['basename'];
+        $dir = 'public://theme-screenshots/' . $payload['repository']['name'];
+        file_prepare_directory($dir, FILE_CREATE_DIRECTORY);
```
[the remark this answers is not in the document]
Comment (klonos) [reply www/modules/custom/borg_github/borg_github.module]: ...or just `screenshots`?

**Comment by docwilmot** [comment]
@jenlampton @quicksketch with the new hook from https://github.com/backdrop-contrib/project/pull/38, we're ready to roll with this again. 

I apologise for any WTFs as I am unable to test this fully since all the GitHub webhooks are set for only the live site so I have to fake webhooks to test.

This PR allows for additional tags in project info files, and maps those tags to fields (already existing) on project nodes on backdropCMS.org. The tags are read during a release. 

Screenshots are special case. Screenshots are needed both to provide large preview images for BackdropCMS (eventually we'd probably have some way to scroll through them there) and also to provide a small image for the Appearance page on a Backdrop site.

The code checks for a screenshots folder, and a `screenshots[]` array in the info file and proceeds as follows:
- if a screenshot.png is found in the theme root, that is used as the small image for the site
- if no screenshot.png is found, the code checks for a `screenshots[]` array in the info file.
- if a `screenshots[default]` is found, thats used as the small image, otherwise the first image in the list is the small image to be used on the live site.
- if no `screenshots[]` array is in the info file, we search for a screenshots folder anyway, and use those images, the first one as the small image, the rest as the large images for BackdropCMS
- failing all that, then both BackdropCMS and the live site would use their default image icon.
- finally the screenshots folder is deleted from the release package (it remains on the githhub repo) so sites dont have to download all of those.

Eventually we'll need code in Installer to show the large images directly from BackdropCMS when a user is browsing themes in Installer.

Other tags managed by this PR:
- `colors[]` (themes)
- `maintainers[]`
- `tags[]`

**Comment by jenlampton** [comment]
> with the new hook from backdrop-contrib/project#38, we're ready to roll with this again.

Has this new hook been deployed yet? 

> if a screenshots[default] is found, thats used as the small image, otherwise the first image in the list is the small image to be used on the live site.

Was this the syntax we agreed on? Seems odd...

> Other tags managed by this PR: ... maintainers[]

I'm almost certain there won't be maintainer information in an info file (since that's located in the readme) ... But I don't remember the results of that other conversation.

Would you please include link to that discussion so I can refresh my memory?

**Comment by docwilmot** [comment]
> Has this new hook been deployed yet?

No, it has not.
> Was this the syntax we agreed on? Seems odd...

See my comment [here](https://github.com/backdrop-ops/backdropcms.org/issues/487#issuecomment-508529053) for the rationale. In summary the small screenshot will end up in the same folder with the larger shots. We need a way to tell BackdropCMS which one it is. The terminology is up for changing, dont mind what we call it.
> Would you please include link to that discussion so I can refresh my memory?

https://github.com/backdrop-ops/backdropcms.org/issues/487#issue-390066164

**Comment by jenlampton** [comment]
This PR was closed accidentally when master branch was deleted. Reopening.

**Pull request closed without merging**